### DISCUSSION TITLE
Enhancement Request for C# Language - Simplified Display of Property Values in Documentation Comments

### DISCUSSION THREAD

**Discussion by Mortens4444** [dotnet/csharplang#7780] [Language Ideas]
I am writing to request an enhancement in the C# language that would simplify the display of property values in documentation comments.

Consider the following class structure:

```csharp
public class Foo : Bar
{
    public Foo()
    {
        Name = "A";
    }
}

public class Foot : Bar
{
    public Foot()
    {
        Name = "B";
    }
}

public abstract class Bar
{
    public string Name;
}
```

I wanted a documentation comment that shows the name value of the class.
So I created this one:

```csharp
/// <summary>
/// A
/// </summary>
public class Foo : Bar
{
    public Foo()
    {
        Name = "A";
    }
}

/// <summary>
/// B
/// </summary>
public class Foot : Bar
{
    public Foot()
    {
        Name = "B";
    }
}

public abstract class Bar
{
    public string Name;
}
```

But this would be much better.

```csharp
/// <inheritdoc/>
public class Foo : Bar
{
    public Foo()
    {
        Name = "A";
    }
}

/// <inheritdoc/>
public class Foot : Bar
{
    public Foot()
    {
        Name = "B";
    }
}

/// <summary>
/// {Name}
/// </summary>
public abstract class Bar
{
    public string Name;
}
```

What is your opinion? Or even without the inheritdoc, much better.

**Comment by TahirAhmadov** [comment]
That's not how comments work - what value a property is assigned cannot be referenced in any way by the comments XML.

**Reply by colejohnson66** [reply]
Perhaps there's an argument that could be made to allow `inheritdoc` to pass in some "props", even if just textual? Something like `<inheritdoc name="A" />`.

**Comment by huoyaoyuan** [comment]
If you want that to work, you need to at least create a constant for the value to be assigned, the request some syntax to embed a constant.
